### PR TITLE
Fix parsing some special chars in depfiles

### PR DESCRIPTION
**Pull request by maciej-pawlowski**
This allows paths with "[", "]" and "%" to appear in depfiles.
Previously, only "[" would be handled properly.
Fixes issue #1227 .

### CODE REVIEW

**Comment by maciej-pawlowski** [comment]
Can I have any feedback whether this is going to get integrated?

**Comment by ppluciennik** [comment]
@nico would it be possible to integrate it? Without this fix, ninja detects that something has changed and rebuilds some targets (It's reproducible on Opera's code - Chromium's based code).

File: src/lexer.cc [hunk 1]
```diff
@@ -1,4 +1,4 @@
-/* Generated by re2c 0.13.5 */
+/* Generated by re2c 0.16 */
```
Comment (jhasse) [inline src/lexer.cc]: Please remove this file from your commit.

File: src/depfile_parser.cc [hunk 2]
```diff
@@ -1,4 +1,4 @@
-/* Generated by re2c 0.13.5 */
+/* Generated by re2c 0.16 */
```
Comment (jhasse) [inline src/depfile_parser.cc]: Can you either generate the file with re2c 0.13.5 or add a separate commit which updates the file in Git from re2c 0.13.5 to 0.16?

**Comment by jhasse** [comment]
Thanks for the PR and sorry we didn't look at this earlier!

I think the merge conflict is due to https://github.com/ninja-build/ninja/commit/265a6eaf399778c746c7d2c02b8742f3c0ff07e9.

**Comment by maciej-pawlowski-opera** [comment]
I've since lost access to that account due to my stupidity and 2FA working as intended...
I've rebased the changes onto current master and split regenerating re2c outputs into separate commits here, in a new fork:
https://github.com/maciej-pawlowski-opera/ninja/tree/issue-1227-2

Should I make a new pull request? I'm not sure how github can handle such a situation more gracefully.

**Comment by jhasse** [comment]
Yes, please just create a new PR :) Add "See #1228" in the description so that it references this one.

**Pull request closed without merging**